kyo-prelude/shared/src/main/scala/kyo/Abort.scala

```diff
@@ -603,4 +603,61 @@ object Abort:
 
     end literal
 
+    sealed class CanMaskAbort
+
+    /** A function to apply a mask to any failures of type [[E]] than may or may not be possible for a generic error effect [[Abort[EG]]].
+      */
+    sealed abstract class MaskAbort[E, EG]:
+        /* Masked error type */
+        type Masked <: Product
+
+        /** @param effect
+          *   Effect including [[Abort[EG]]] which may or may not include error subtype [[E]]
+          * @return
+          *   An effect with any [[Abort[E]]] failures masked as [[Masked]]
+          */
+        def apply[A, S](effect: A < (Abort[EG] & S)): A < (Abort[EG | Masked] & S)
+
+        given tag: ConcreteTag[Masked]
```

Comment:
it seems this doesn't need to be in the abstract class?

Reply:
I couldn't make it work without it here. If you need to call `Abort.run[mask.Masked]` within a `withMask` effect, this is the only way I can figure out to make the `ConcreteTag` available to the compiler.